### ISSUE TITLE
[valibot] Validation errors reported by ‘checkItems’ are not displayed properly

### ISSUE THREAD

**Issue by Thanaen** [react-hook-form/resolvers#707]
**Describe the bug**
Validation errors reported by ‘checkItems’ should be displayed in the same way as other errors

**To Reproduce**

1. Focus the first input
2. Insert "t"
3. Submit (to trigger a length error)
4. Update the input value to "value"
5. Add another input using the "Add input" button
6. Insert "value" in the newly added input
7. Submit

**Codesandbox link (Required)**
https://stackblitz.com/edit/vitejs-vite-yj5g5r?file=src%2FApp.tsx

**Expected behavior**
Validation errors reported by ‘checkItems’ should be displayed in the same way as other errors

Form schema:
```ts
const FormSchema = v.object({
  test: v.pipe(
    v.array(
      v.object({
        value: v.pipe(v.string(), v.minLength(4)),
      })
    ),
    v.checkItems(
      (item, index, values) =>
        values.map((v) => v.value).indexOf(item.value) === index,
      'Duplicate items'
    )
  ),
});
```

Error displayed properly:
![image](https://github.com/user-attachments/assets/3fc5dacd-17db-4412-8a47-fd19b93af736)

Error not displayed properly:
![image](https://github.com/user-attachments/assets/6fcc2aa9-dda5-4e13-a85a-5d42166db4ed)

**Additional context**
```json
"dependencies": {
    "@hookform/resolvers": "^3.9.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-hook-form": "^7.52.1",
    "valibot": "^0.36.0"
  }
```

**Comment by jorisre** [comment]
It appears there’s an issue (or misuse) with Valibot when combining `forward` and `checkItems` within a `v.array()`. 

The problem lies in the incorrect path, which causes the resolver to return an invalid path, preventing the error message from being displayed.

Does this clarify things for you?

👉🏻 https://codesandbox.io/p/sandbox/q8krhm

**Comment by Thanaen** [comment]
Indeed, it's not a problem with the resolver, but simply a problem in my approach. I'm closing the issue!
